### PR TITLE
cmd/importer: support probability in building incremental column

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve? 

Generate column values in incremental order mostly, but with some exceptions controlled by input parameter.

### What is changed and how it works?

Support new rule `probability=xxx` which should be used together with `incremental`, to control the exceptions in overall incremental order.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

```
run script:
./bin/importer -t "create table t(a date comment '[[incremental=1;repeats=3;step=-1;probability=80]]');" -c 1 -n 10 -P 4000

check table rows:
mysql> select * from t;
+------------+
| a          |
+------------+
| 2019-05-14 |
| 2019-05-02 |
| 2019-05-13 |
| 2019-06-05 |
| 2019-05-13 |
| 2019-05-12 |
| 2019-05-12 |
| 2019-08-26 |
| 2019-05-11 |
| 2019-05-11 |
+------------+
10 rows in set (0.00 sec)
```

Code changes

N/A

Side effects

N/A

Related changes

N/A

### CODE REVIEW

File: cmd/importer/README.md [hunk 1]
```diff
@@ -166,9 +166,11 @@ MySQL [test]> select * from t;
 10 rows in set (0.002 sec)
 ```
 
+`probability` controls the exceptions of `incremental` and `repeats`.
```
Comment (lamxTyler) [inline cmd/importer/README.md]: Explain the relationship between the amount of `exception` and `probability`? Seems that the more probability, the fewer exceptions.

File: cmd/importer/db.go [hunk 2]
```diff
@@ -136,13 +137,24 @@ func genColumnData(table *table, column *column) (string, error) {
 	if _, ok := table.uniqIndices[column.name]; ok {
 		incremental = true
 	}
+	if incremental {
+		incremental = uint32(rand.Int31n(100))+1 <= column.data.probability
+		// If incremental, there is only one worker, so it is safe to directly access datum.
```
Comment (lamxTyler) [inline cmd/importer/db.go]: But `incremental` will also be true when this column is `unique`, in this case, there may be other workers.

Comment (eurekaka) [reply cmd/importer/db.go]: Updated, PTAL

File: cmd/importer/data.go [hunk 3]
```diff
@@ -32,34 +31,30 @@ type datum struct {
...
-		d.intValue = min
-	}
-
-	if min < max {
-		d.maxIntValue = max
-		d.useRange = true
+	d.minIntValue = min
+	d.maxIntValue = max
+	d.useRange = true
+	if d.step < 0 {
```
Comment (zz-jason) [inline cmd/importer/data.go]: should we specially handle 0 step?

File: cmd/importer/data.go [hunk 4]
```diff
@@ -75,8 +70,13 @@ func (d *datum) nextInt64() int64 {
 	}
 	if d.useRange {
 		d.intValue = mathutil.MinInt64(d.intValue, d.maxIntValue)
+		d.intValue = mathutil.MaxInt64(d.intValue, d.minIntValue)
+	}
+	if uint32(rand.Int31n(100))+1 <= 100-d.probability {
```
Comment (zz-jason) [inline cmd/importer/data.go]: how about extract a common function to decide the next remains? for example:
```go
func (d *datum) refreshRemains() {
	if uint32(rand.Int31n(100))+1 <= 100-d.probability {
		d.remains -= uint64(rand.Int63n(int64(d.remains))) + 1
	} else {
		d.remains--
	}
}
```

File: cmd/importer/data.go (hunk 3 above)
Comment (eurekaka) [reply cmd/importer/data.go]: If step is 0, we can use `range` or `set` rule, instead of `incremental`?

**Comment by XuHuaiyu** [comment]
please **IGNORE** this comment
/run-all-tests tidb-test=pr/814

File: cmd/importer/data.go (hunk 3 above)
Comment (zz-jason) [reply cmd/importer/data.go]: @lamxTyler How do you think?

Comment (lamxTyler) [reply cmd/importer/data.go]: 0 step also makes senses, for example, we only want constant data.

Comment (eurekaka) [reply cmd/importer/data.go]: Yes, but if we want constant data, we don't need to use `incremental`, a better way is using `set`?

Comment (zz-jason) [reply cmd/importer/data.go]: I prefer to do this optimization in the future, for now, let's make it work.

**Pull request merged**